### PR TITLE
Add new event to be able to notify group creation.

### PR DESCRIPTION
**Pull request by gbastien**
Hi,

I created a new event "GroupCreated" and notify it in ZODBGroupManager.addGroup and DynamicGroupsPlugin.addGroup.

Could you please review this?  I will also make a PR for 1.x branch as we need a new release for Zope2.

Thank you!

Gauthier

### CODE REVIEW

File: Products/PluggableAuthService/plugins/DynamicGroupsPlugin.py
Comment (tseaver) [inline Products/PluggableAuthService/plugins/DynamicGroupsPlugin.py:337]: Needs new test assertion that the event is emitted.

File: Products/PluggableAuthService/plugins/ZODBGroupManager.py
Comment (tseaver) [inline Products/PluggableAuthService/plugins/ZODBGroupManager.py]: Needs new test assertion that the event is emitted.

**Review by tseaver — COMMENTED** [review]
In addition to the new tests, please hold of on the 1.x branch PR until we get PR #21 landed.

File: Products/PluggableAuthService/plugins/DynamicGroupsPlugin.py
Comment (gbastien) [reply Products/PluggableAuthService/plugins/DynamicGroupsPlugin.py:337]: Hi @tseaver 

how can I do this? I can either register a testing subscriber that "does something", but I suppose there is another way to check this, do you have sample test somewhere?

Thank you for feedback,

Gauthier

Comment (icemac) [reply Products/PluggableAuthService/plugins/DynamicGroupsPlugin.py:337]: This could be an idea to write such a test.

**Review by icemac — CHANGES_REQUESTED** [review]
LGTM, but it should have a test.

File: Products/PluggableAuthService/plugins/DynamicGroupsPlugin.py
Comment (gbastien) [reply Products/PluggableAuthService/plugins/DynamicGroupsPlugin.py:337]: Hi @tseaver @icemac 

I added a test using similar way as for UserFolder creation.  I finally had to add the group management plugin instance to the called event.  First it was to do the test work...  But finally I think it is useful to have a persistent object passed to the event or we only receive a group_id and we do not know where it comes from...

Please review and tell me.

Gauthier

File: Products/PluggableAuthService/plugins/tests/test_ZODBGroupManager.py
Comment (icemac) [inline Products/PluggableAuthService/plugins/tests/test_ZODBGroupManager.py]: This should be done in `setUp` and undone in `tearDown`.

**Review by icemac — CHANGES_REQUESTED** [review]
LGTM besides one request to change.

File: Products/PluggableAuthService/plugins/tests/test_ZODBGroupManager.py
Comment (gbastien) [reply Products/PluggableAuthService/plugins/tests/test_ZODBGroupManager.py]: @icemac mimics test_UserFolder.UserEvents.testUserCreationEvent...

I adapted code, thank you for review.

Gauthier

**Review by icemac — APPROVED** [review]

**Comment by gbastien** [comment]
Hi @icemac @tseaver 

when merged, I will propose the same PR for the 1.x branch.

Thank you for feedback,

Gauthier

**Comment by icemac** [comment]
@gbastien As you are a Zope committer you should be able to push the big merge button yourself.

**Pull request merged**
